crates/pathfinder/src/config.rs

```diff
@@ -129,6 +143,76 @@ impl Configuration {
 
         let cfg = cfg.try_build()?;
 
+        if cfg.integration && cfg.testnet2 {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                "Cannot use both integration and testnet 2 at the same time. Please choose one."
+                    .to_string(),
+            ));
+        }
+
+        if cfg.gateway.is_some() && cfg.sequencer_url.is_some() {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                "Cannot use both gateway and sequencer-url at the same time. Please use gateway only."
+                    .to_string(),
+            ));
+        }
+
+        if cfg.network.is_some() && cfg.integration {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                "Cannot specify both network and integration options at the same time. Please use network only."
+                    .to_string(),
+            ));
+        }
+
+        if cfg.network.is_some() && cfg.testnet2 {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidInput,
+                "Cannot specify both network and testnet2 options at the same time. Please use network only."
+                    .to_string(),
+            ));
+        }
+
+        // Emit warning logs for deprecated configuration options.
+        // TODO: remove warnings and the options in next major release.
+        if cfg.sequencer_url.is_some() {
+            tracing::warn!("'--sequencer-url' is deprecated, please use '--gateway' instead");
+        }
+
+        if cfg.integration {
+            tracing::warn!(
+                "'--integration' is deprecated, please use '--network integration' instead"
+            );
+        }
+
+        if cfg.testnet2 {
+            tracing::warn!("'--testnet2' is deprecated, please use '--network testnet2' instead");
+        }
+
+        // Emit warning if network option is not provided.
+        // TODO: remove warning and make option required in next major release.
+        if cfg.network.is_none() {
+            tracing::warn!("'--network' configuration not specified - this will be a required item in a future version of pathfinder.");
```

Comment:
Do we want to make `--network` mandatory? I thought the concept of _defaulting_ to mainnet/testnet based on the Ethereum chain was a good one. (And supposing that testnet2 is a temporary thing...)

Reply:
I'm open for debate - I went back and forth on this a few times.

It becomes a bit.. tricky because we also have proxy gateway urls and then also p2p networks coming up. I thought it might be safer to have them specify exactly what it is that they want.

Reply:
I was just suggesting to keep the defaults so that we don't break our CLI. And in general I think these defaults just make sense for most people running pathfinder.